lms/djangoapps/program_enrollments/api/v1/views.py

```diff
@@ -78,6 +78,44 @@ def wrapped_function(self, request, **kwargs):
     return wrapped_function
 
 
+def verify_course_exists_and_in_program(view_func):
+    """
+    Raises:
+        An api error if the course run specified by the `course_key` kwarg
+        in the wrapped function is not part of the curriculum of the program
+        specified by the `program_uuid` kwarg
+
+    Assumes that the program exists
+    """
+    @wraps(view_func)
+    @verify_course_exists
+    def wrapped_function(self, request, **kwargs):
+        """
+        Wraps view function
+        """
+        course_key = CourseKey.from_string(kwargs['course_id'])
+        program_uuid = kwargs['program_uuid']
+        program = get_programs(uuid=program_uuid)
+        active_curricula = (c for c in program['curricula'] if c['is_active'])
+        curriculum = next(active_curricula)
```

Comment:
I wasn't sure whether or not to include some sort fo check here for multiple active curricula. What do y'all think?

Reply:
I'd just leave a comment about the assumption, that's what we did in registrar.